src/pages/academy/components/search_banner/_search-bar.tsx

```diff
@@ -0,0 +1,419 @@
+                                : {}
+                        }
+                    >
+                        {part}
+                    </span>
+                ))}
+            </span>
+        )
+    }
+
+    const onClickIcon = () => {
+        navigate(`/academy/search?q=${encodeURI(search_input)}`)
+        setSearchInput('')
+        is_mobile_separator && setModal(false)
+    }
+
+    const getResultTitles = (item) =>
+        item.blog_title
+            ? getHighlightedTitle(item.blog_title, search_query)
+            : getHighlightedTitle(item.video_title, search_query)
+
+    useOutsideClick(search_bar_ref, () => handleBlur())
+
+    return (
+        <>
+            <FormWrapper fd="column" ai="flex-start" height="auto">
+                <FormContainer
+                    role="search"
+                    aria-label="Academy"
+                    spellCheck={false}
+                    onSubmit={handleSubmit}
+                >
+                    <FlexSearchBar
+                        jc="flex-start"
+                        ai="center"
+                        is_maximise={search_input_touched}
+                        result_opened={result_opened}
+                        ref={search_bar_ref}
+                    >
+                        <SearchIconWrapper
+                            src={SearchIcon}
+                            alt="search_icon"
+                            onClick={onClickIcon}
+                        ></SearchIconWrapper>
+                        <InputWrapper
+                            type="text"
+                            placeholder="I want to know about..."
+                            onChange={handleFilterSearch}
+                            onFocus={handleFocus}
+                            value={search_input}
+                            ref={input_ref}
+                            onKeyDown={handleNavigation}
+                        ></InputWrapper>
+                        <CloseIconWrapper
+                            is_maximise={search_input_touched}
+                            src={CloseIcon}
+                            alt="close icon"
+                            onClick={handleBlur}
+                        />
+                    </FlexSearchBar>
+                </FormContainer>
+                {search_query && (
+                    <DesktopWrapper>
+                        <SearchSuggestionWrapper max_width="100%" fd="column">
+                            <Line />
+                            <SearchResultRows
+                                jc="flex-start"
+                                ai="center"
+                                is_active={focus_index === -1}
+                                style={{ color: 'var(--color-blue-3)' }}
+                                onMouseDown={handleSearchClick}
+                            >{`Search for: ${search_query}`}</SearchResultRows>
+
+                            {data_to_render &&
```

Comment:
what do you think about this? 
```suggestion
                            {data_to_render?.slice(0, 6)?.map(...)
```
I think it's gonna be cleaner this way, but if you don't like it ignore this comment. :pray: